### PR TITLE
Feat/user photo upload 2

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

File: app/api/user/photo/route.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { put } from '@vercel/blob'
+import { NextResponse } from 'next/server'
+
+export async function POST(request: Request): Promise<NextResponse> {
+  const { searchParams } = new URL(request.url)
+  const filename = searchParams.get('filename')
+
+  if (!filename || !request.body) {
+    return new NextResponse('Error: Invalid filename or request body.', {
```
Comment (Sumick) [inline app/api/user/photo/route.ts:9]: **nitpick:** If it's a generic file upload route, we should keep it in `app/api/files/route.ts`

File: src/components/CreateProfile/CreateProfileTopBar/CreateProfileTopBar.tsx [hunk 2]
```diff
@@ -12,12 +12,12 @@ const CreateProfileTopBar = () => {
...
 
-  const handleButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
-    runAsync(async () => {
-      event.preventDefault()
-      handleSubmit()
-      router.push(AppRoutes.myProfile)
-    })
+  const handleButtonClick = async (
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
```
Comment (Sumick) [inline src/components/CreateProfile/CreateProfileTopBar/CreateProfileTopBar.tsx:17]: **nitpick:** Why `runAsync` is removed here?

File: src/components/EditProfileForm/EditProfileFormWrapper.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import { mapProfileModelToEditProfileFormValues } from '@/components/EditProfile
 
 export interface EditProfileFormValues {
   fullName: string
+  avatarUrl: string | null
```
Comment (Sumick) [inline src/components/EditProfileForm/EditProfileFormWrapper.tsx:13]: **nitpick:** `avatarUrl` should not be optional

File: src/components/UserPhotoUploader/UserPhotoUploader.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  const { data: session } = useSession()
+  const importFromGithub = async () => {
+    setUserImage(session?.user?.image || '')
+  }
+
+  const updateUserAvatar = async (avatarUrl: string) => {
+    try {
+      const response = await fetch('/api/user', {
```
Comment (Sumick) [inline src/components/UserPhotoUploader/UserPhotoUploader.tsx:34]: **nitpick:** this should be a part of `apiClient`

File: app/api/user/route.ts [hunk 5]
```diff
@@ -40,3 +41,16 @@ export async function POST(request: NextRequest) {
     return new NextResponse(`${error}`, { status: 500 })
   }
 }
+
+export async function PATCH(request: NextRequest) {
+  try {
```
Comment (Sumick) [inline app/api/user/route.ts:46]: **nitpick:** This handler is image upload specific, it should be deeper - `api/user/avatar/route.ts`. Or if you want to leave it here, you'd have to set whole user update possible, not just avatar:
```ts
const updatedUser = await updateUser(email, payload)
```

File: src/components/UserPhotoUploader/UserPhotoUploader.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      reader.onloadend = () => {
+        setUserImage(reader.result as string)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const { data: session } = useSession()
+  const importFromGithub = async () => {
+    setUserImage(session?.user?.image || '')
```
Comment (Sumick) [inline src/components/UserPhotoUploader/UserPhotoUploader.tsx:29]: **nitpick:** We should check `if (!sessionImage) throw Error("No Github image found")`

File: src/components/UserPhotoUploader/UserPhotoUploader.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      console.log('Updated user:', data)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const handleUpload = async () => {
+    if (selectedFile) {
+      fetch('/api/user/photo', {
+        method: 'POST',
```
Comment (Sumick) [inline src/components/UserPhotoUploader/UserPhotoUploader.tsx:56]: **nitpick:** This should also be in `apiClient`

File: src/components/UserPhotoUploader/UserPhotoUploader.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+      const reader = new FileReader()
+      reader.onloadend = () => {
+        setUserImage(reader.result as string)
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const { data: session } = useSession()
+  const importFromGithub = async () => {
```
Comment (Sumick) [inline src/components/UserPhotoUploader/UserPhotoUploader.tsx:28]: **issue:** I don't see it's updating user image in database

**Review by Sumick — CHANGES_REQUESTED** [review]
This approach looks nice! We need to clean it a bit, and improve maintainability, but we are really close

File: src/components/UserPhotoUploader/UserPhotoUploader.tsx (hunk 8 above)
Comment (wawrzonnn) [reply src/components/UserPhotoUploader/UserPhotoUploader.tsx:28]: me too

**Pull request closed without merging**